dash-renderer/src/actions/setAppReadyState.js

```diff
@@ -35,43 +41,47 @@ const isAppReady = layout => {
         }
     }
 
-    const promises = [];
-    Object.entries(res).forEach(([namespace, item]) => {
-        Object.entries(item).forEach(([type]) => {
-            const component = Registry.resolve({
-                namespace,
-                type,
+    return targets => {
+        const promises = [];
+
+        if (Array.isArray(targets)) {
+            targets.forEach(id => {
+                const target = ids[id];
+                if (target) {
+                    const component = Registry.resolve(target);
+
+                    const ready = isReady(component);
+
+                    if (ready && typeof ready.then === 'function') {
+                        promises.push(ready);
+                    }
+                }
             });
+        } else {
+            Object.entries(components).forEach(([namespace, item]) => {
+                Object.entries(item).forEach(([type]) => {
+                    const component = Registry.resolve({
+                        namespace,
+                        type,
+                    });
 
-            const ready = isReady(component);
+                    const ready = isReady(component);
 
-            if (ready && typeof ready.then === 'function') {
-                promises.push(ready);
-            }
-        });
-    });
+                    if (ready && typeof ready.then === 'function') {
+                        promises.push(ready);
+                    }
+                });
+            });
+        }
 
-    return promises.length ? Promise.all(promises) : true;
+        return promises.length ? Promise.all(promises) : true;
+    };
 };
```

Comment:
Change `isAppReady` to return either the global state of the app or the state for the target ids given.